applications/ShapeOptimizationApplication/custom_utilities/response_functions/water_drain_response_function_utility.h

```diff
@@ -0,0 +1,184 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Schmölz David, https://github.com/dschmoelz
+//
+// ==============================================================================
+
+#ifndef WATER_DRAIN_FUNCTION_UTILITY_H
+#define WATER_DRAIN_FUNCTION_UTILITY_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "custom_utilities/geometry_utilities.h"
+
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+typedef Node NodeType;
+typedef NodeType::Pointer NodeTypePointer;
+typedef std::vector<NodeType::Pointer> NodeVector;
+typedef ModelPart::NodesContainerType NodesArrayType;
+
+struct Volume {
+	IndexType Id;
+	IndexType mLowPointId;
+	Vector mLowestPoint;
+	double mMaxWaterLevel = 0;
+	bool isMerged = false;
+	std::vector<int> mNeighbourVolumes;
+};
+
+/// Short class definition.
+/** Detail class definition.
+
+ */
+
+class KRATOS_API(SHAPE_OPTIMIZATION_APPLICATION) WaterDrainResponseFunctionUtility
+{
+public:
+	///@name Type Definitions
```

Comment:
it would be nice to add some description and documentation to the class.

Reply:
I've added some descriptions :)